docs/DeveloperGuide.md

```diff
@@ -1,11 +1,267 @@
 # Developer Guide
+**By:** Team T14-4<br/>
+**Since:** May 2020
 
-## Design & Implementation
+* [1. Design & Implementation](#design-implementation)
+    + [1.1. [Proposed] Search stock feature](#search-stock)
+    + [1.2. [Proposed] List stock in descending quantities](#list-stock)
+    + [1.3. [Proposed] Generate profit for the day](#generate-profit)
+    + [1.4. [Proposed] Search reservation](#search-reservation)
+    + [1.5. [Proposed] Search dish](#search-dish)
+* [2. Product Scope](#product-scope)
+* [3. User Stories](#user-stories)
+* [4. Non-Functional Requirements](#nonfunctional-requirement)
+* [5. Glossary](#glossary)
+
+
+<a name="design-implementation"></a>
+
+## 1. Design & Implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+<a name="search-stock"></a>
+
+### 1.1 [Proposed] Search stock feature
+#### 1.1.1 Proposed implementation
+
+In the restaurant daily report, users can search against the stock category by supplying a keyword.
+
+Given below is an example usage scenario and how the search mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. An empty `stock` will be initialized.
+
+Step 2. The user executes `add stock; i/tomato; q/10; p/$0.40;` command to add a tomato ingredient into the `stock`. Further, the user may add more ingredients into the current `stock`. Suppose the user executes `add stock; i/potato; q/5; p/$0.40;` and `add stock; i/rice; q/3; p/$0.40;` as well.
+
+Step 3. The user can now search against the current `stock` to see if an ingredient is stored in the `stock`. The user now executes `search stock; k/tomato;`, which will display the following result in the image. 
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/59989652/77921374-47b58b00-72d2-11ea-84bd-fe0c790b1d59.PNG">
+</p>
```

Comment:
![searchstock](https://user-images.githubusercontent.com/42376931/77988700-8d646900-734f-11ea-82c0-7751ce275c1e.JPG)
Perhaps you can exclude these first 3 commands in the screenshot and simply explain that we already have a list containing these ingredients. You can then go on to just show the result that search stock will return. This would perhaps make the diagram clearer as after all the focus should be on the search stock command

Reply:
I agree. I'll update my images accordingly.